common/cache/lru_test.go

```diff
@@ -107,6 +107,11 @@ func TestGenerics(t *testing.T) {
 		dummyString: "some other random key",
 		dummyInt:    56,
 	}))
+	assert.Equal(t, 1, cache.Size())
+
+	cache.Put(key, "some other random value")
```

Comment:
Given there was a bug already, I'd like to see a much more through test including calls to Put and PutIfNotExist with duplicate and non-duplicate keys